core/src/main/java/jenkins/views/Header.java

```diff
@@ -0,0 +1,36 @@
+package jenkins.views;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+/**
+ * Extension point that provides capabilities to render a specific header
+ * 
+ * @author Ildefonso Montero
+ * @see JenkinsHeader
+ */
+public abstract class Header implements ExtensionPoint {
+    
+    /**
+     * Checks if header is enabled.
+     * @return if header is enabled
+     */
+    public abstract boolean isHeaderEnabled();
+
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public static Header get() {
+        List<Header> headers = ExtensionList.lookup(Header.class).stream().filter(header -> header.isHeaderEnabled()).collect(Collectors.toList());
```

Comment:
```suggestion
        List<Header> headers = ExtensionList.lookup(Header.class).stream().filter(Header::isHeaderEnabled).collect(Collectors.toList());
```

plus @jtnord’s suggestions

Reply:
Applied in https://github.com/jenkinsci/jenkins/pull/5909/commits/1dd8d32e52489255cc7e91e25b226be138995f82